### ISSUE TITLE
DL1000 triggered when using `--mount=type=secret,id=my-secret,env=SECRET`

### ISSUE THREAD

**Issue by gustavovalverde** [hadolint/hadolint#1083]
Docker allows mounting secrets and assign them to an environment variable instead of a file by using `env=SECRET`, but hadolint fails when using this approach, e.g.:

```dockerfile
RUN --mount=type=secret,id=aws-key-id,env=AWS_ACCESS_KEY_ID \
    --mount=type=secret,id=aws-secret-key,env=AWS_SECRET_ACCESS_KEY \
    --mount=type=secret,id=aws-session-token,env=AWS_SESSION_TOKEN \
    aws s3 cp ...
```

Fails with:

```shell
unexpected "env=AWS_ACCESS"
expecting "gid=", "id=", "mode=", "required", "required=False", "required=True", "required=false", "required=true", "uid=", source=, or target= (Trunk)hadolint[DL1000](https://github.com/hadolint/hadolint/wiki/DL1000)
```

Reference: https://docs.docker.com/build/building/secrets/#target

**Comment by loozhengyuan** [comment]
Duplicates #1039 and resolved in https://github.com/hadolint/language-docker/pull/100. I _think_ we are just waiting for a new release to be cut.
